pkg/cmd/repo/create/create.go

```diff
@@ -349,90 +333,324 @@ func createRun(opts *CreateOptions) error {
+	if err != nil {
+		return err
+	}
+
+	if opts.Interactive {
+		opts.Source, err = interactiveSource()
+		if err != nil {
+			return err
+		}
+	}
+
+	repoPath := opts.Source
+
+	var baseRemote string
+	if opts.Remote == "" {
+		baseRemote = "origin"
+	} else {
+		baseRemote = opts.Remote
+	}
+
+	projectDir, projectDirErr := git.GetDirFromPath(repoPath)
+	if projectDirErr != nil {
+		return projectDirErr
+	}
+	if projectDir != ".git" {
+		return fmt.Errorf("path not a local repository")
+	}
+
+	abs, err := filepath.Abs(repoPath)
+	if err != nil {
+		return err
+	}
+
+	if opts.Interactive {
+		opts.Name, opts.Description, opts.Visibility, err = interactiveRepoInfo(filepath.Base(abs))
+		if err != nil {
+			return err
+		}
+	}
+
+	var repoToCreate ghrepo.Interface
+
+	// repo name will be currdir name or specified
+	if opts.Name == "" {
+		repoToCreate = ghrepo.NewWithHost("", filepath.Base(abs), host)
+	} else if strings.Contains(opts.Name, "/") {
+		var err error
+		repoToCreate, err = ghrepo.FromFullName(opts.Name)
+		if err != nil {
+			return fmt.Errorf("argument error: %w", err)
+		}
+	} else {
+		repoToCreate = ghrepo.NewWithHost("", opts.Name, host)
+	}
+
+	input := repoCreateInput{
+		Name:              repoToCreate.RepoName(),
+		Visibility:        opts.Visibility,
+		OwnerLogin:        repoToCreate.RepoOwner(),
+		TeamSlug:          opts.Team,
+		Description:       opts.Description,
+		HomepageURL:       opts.Homepage,
+		HasIssuesEnabled:  !opts.DisableIssues,
+		HasWikiEnabled:    !opts.DisableWiki,
+		GitIgnoreTemplate: opts.GitIgnoreTemplate,
+		LicenseTemplate:   opts.LicenseTemplate,
+	}
+
+	repo, err := repoCreate(httpClient, repoToCreate.RepoHost(), input)
+	if err != nil {
+		return err
+	}
+
+	if isTTY {
+		fmt.Fprintf(stdout,
+			"%s Created repository %s on GitHub\n",
+			cs.SuccessIconWithColor(cs.Green),
+			ghrepo.FullName(repo))
+	}
+
+	protocol, err := cfg.Get(repo.RepoHost(), "git_protocol")
+	if err != nil {
+		return err
+	}
+
+	remoteURL := ghrepo.FormatRemoteURL(repo, protocol)
+
+	if opts.Interactive {
+		var addRemote bool
+		remoteQuesiton := &survey.Confirm{
+			Message: `Add a remote?`,
+			Default: true,
+		}
+		err = prompt.SurveyAskOne(remoteQuesiton, &addRemote)
+		if err != nil {
+			return err
+		}
+
+		if !addRemote {
+			return nil
+		}
+
+		pushQuestion := &survey.Input{
+			Message: "What should the new remote be called?",
+			Default: "origin",
+		}
+		err = prompt.SurveyAskOne(pushQuestion, &baseRemote)
+		if err != nil {
+			return err
+		}
+	}
+
+	//get the current branch
+	currentBranch, err := git.CurrentBranch()
+	if err != nil {
+		//How do we wanna handle if there is no current branch?
+		return err
+	}
+
+	if err := sourceInit(opts.IO, remoteURL, baseRemote, repoPath, currentBranch); err != nil {
+		return err
+	}
+
+	if opts.Interactive {
+		pushQuestion := &survey.Confirm{
+			Message: fmt.Sprintf(`Would you like to push local commits on "%s" and track "%s"?`, currentBranch, baseRemote),
```

Comment:
Let's assume that the user wants tracking by default and not ask them about that specifically. We can ask "Would you like to push commits from the current branch to the remote?"